### PR TITLE
Ipaddr2 test deployment wait VM to run

### PR DESCRIPTION
**Pull request by mpagot**
Add waiting function at the end of the deployment.

- Related ticket: https://jira.suse.com/browse/TEAM-9428

# Verification run:

 - sle-15-SP5-SapCloud-Azure-Byos-x86_64-BuildLATEST_AZURE_SLE15_5_BYOS-ipaddr2_azure_test@64bit -> http://openqaworker15.qa.suse.cz/tests/287686

### CODE REVIEW

**Comment by badboywj** [comment]
I am not familiar with the part about unit test, but LGTM about  the lib part in this PR

File: lib/sles4sap/azure_cli.pm [hunk 1]
```diff
@@ -667,6 +668,55 @@ sub az_vm_instance_view_get {
...
+        timeout => 300)
+
+Get the VM state until status looks like:
+
+[
+  "PowerState/running",
+  "VM running"
+]
+
+or reach timeout. Polling frequency is dinamically calculated based on the timeout
```
Comment (lpalovsky) [inline lib/sles4sap/azure_cli.pm:685]: s/dinamically/dynamically/

File: lib/sles4sap/azure_cli.pm [hunk 2]
```diff
@@ -667,6 +668,55 @@ sub az_vm_instance_view_get {
...
+
+=back
+=cut
+
+sub az_vm_wait_running {
+    my (%args) = @_;
+    foreach (qw(resource_group name)) {
+        croak("Argument < $_ > missing") unless $args{$_}; }
+    $args{timeout} //= 300;
+    my $sleep_time = $args{timeout} < 60 ? int($args{timeout} / 2) : 30;
```
Comment (lpalovsky) [inline lib/sles4sap/azure_cli.pm:703]: I would suggest adding a comment with explanation for this formula.

**Review by lpalovsky — APPROVED** [review]
LGTM

**Review by lilyeyes — APPROVED** [review]
LGTM.

**Pull request merged**